File: backend/api/dto/secret.go
package dto

type Secret struct {
	Name      string `json:"name"`
	Value     string `json:"value"`
	Note      string `json:"note,omitempty"`
	Protected bool   `json:"protected,omitempty"`
}

type SecretRequestDto struct {
	Secrets []Secret `json:"secrets"`
}

File: frontend/app/models/secret.ts
import { z } from "zod";
import { ServerResponse, ServerResponseSchema } from "./default";

export const SecretSchema = z.object({
  name: z.string(),
  value: z.string(),
  note: z.string().optional(),
  protected: z.boolean().optional(),
});

export type Secret = z.infer<typeof SecretSchema>;

export const SecretsSchema = z.array(SecretSchema);

export type Secrets = z.infer<typeof SecretsSchema>;

export const GetSecretsSchema = ServerResponseSchema.extend({
  data: SecretsSchema,
});

export type GetSecrets = z.infer<typeof GetSecretsSchema>;
